Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -4934,13 +5002,15 @@ template <class TreeBuilder> TreeExpression Parser<LexerType>::parsePrimaryExpre
         return context.createThisExpr(location);
     }
     case AWAIT:
+        semanticFailIfTrue(currentScope()->isStaticBlockScope(), "The 'await' keyword is disallowed in the IdentifierReference position features within class static block scope");
         if (m_parserState.functionParsePhase == FunctionParsePhase::Parameters)
             semanticFailIfFalse(m_parserState.allowAwait, "Cannot use 'await' within a parameter default expression");
         else if (currentFunctionScope()->isAsyncFunctionBoundary() || isModuleParseMode(sourceParseMode()))
             return parseAwaitExpression(context);
 
         goto identifierExpression;
     case IDENT: {
+        failIfTrue(isArgumentsToken() && !findClosestFunctionScopeUntilStaticBlock(), "Class static block cannot use 'arguments' as an identifier");
```

Comment:
This `findClosestFunctionScopeUntilStaticBlock` too costly since this is one of the hottest path in parser. See comment around `findClosestFunctionScopeUntilStaticBlock`.

Reply:
Because we already matched `IDENT`, we should do `*m_token.m_data.ident == m_vm.propertyNames->arguments`